design/object-store-realm.md

```diff
@@ -0,0 +1,214 @@
+# Rook Object Store Realm
+
+## Overview
+
+Multisite replication between object stores can be configured to replicate the data in a realm over multiple Ceph clusters. 
+
+Multisite also allows object stores to be independent and isolated from other object stores in a cluster. This is done by implementing multiple independent realms in a single Rook Ceph cluster.
+
+Multiple independent realms in a single Rook Ceph cluster can be replicated over multiple Ceph clusters.
+
+### Prerequisites
+
+When a zone is being created on another Ceph cluster, an RGW in the master zone must have an external endpoint exposed that can be reached by the other cluster.
+
+### Ceph multi-site object store data model
+
+For reference, here is a description of the underlying Ceph Multisite data model.
+
+```
+A cluster has one or more realms
+
+A realm spans one or more clusters
+A realm has one or more zonegroups
+A realm has one master zonegroup
+A realm defined in another cluster is replicated with the pull command
+
+A zonegroup has one or more zones
+A zonegroup has one master zone
+A zonegroup spans one or more clusters
+A zonegroup defined in another cluster is replicated with the pull command
+A zonegroup defines a namespace for object IDs unique across its zones
+Zonegroup metadata is replicated to other zonegroups in the realm
+
+A zone belongs to one cluster
+A zone has a set of pools that store the user and object metadata and object data
+Zone data and metadata is replicated to other zones in the zonegroup
+A master zone needs to be created for secondary zones to pull from to replicate across zones
+```
+
+## Object Store Multisite Walkthrough
+
+### Creating a Realm & Zonegroups
+
+#### Config
+
+When the storage admin is ready to create a realm and zonegroups for object storage, the admin will specify his/her desired settings in the Realm section on the configuration file.
+
+This example `object-store-realm.yaml`, names a realm with 2 zonegroups.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph
+spec:
+  bootstrap-realm:
+    name: my-realm
+    zonegroups:
+      my-zonegroup-a:
+        is-master: true
+        master-zone: zone-z
+      my-zonegroup-b:
+        is-master: false
```

Comment:
nit: if we do keep the properties, camel casing is preferred instead of naming with dashes. For example `isMaster` and `masterZone`. However, the values can have dashes.